### PR TITLE
perf: Change current directory in public/index.php only if necessary

### PR DESCRIPTION
**Pull request by BennyBPB**
**Description**
Checked if current directory is pointing to the front controller's directory.
Only change the directory if they differ because chdir call is not necessary and is not performant.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you for sending this PR.

First of all, you must sign all your commits.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#signing

**Comment by kenjis** [comment]
The `chdir()` came from: #36

**Comment by kenjis** [comment]
> Only change the directory if they differ because chdir call is not necessary and is not performant.

Can you show the benchmarks?

**Comment by BennyBPB** [comment]
> 
Running on Windows 10 with XAMPP
Loop 100000 times:

chdir(FCPATH);
4.0267419815063 seconds

if (getcwd() . DIRECTORY_SEPARATOR !== FCPATH) {}
0.030242919921875 seconds

**Comment by BennyBPB** [comment]
> Thank you for sending this PR.
> 
> First of all, you must sign all your commits. See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#signing

Done

**Comment by BennyBPB** [comment]
> > Only change the directory if they differ because chdir call is not necessary and is not performant.
> 
> Can you show the benchmarks?

https://github.com/codeigniter4/CodeIgniter4/pull/7559#issuecomment-1599294539

**Comment by kenjis** [comment]
Thanks. 

But see https://github.com/codeigniter4/CodeIgniter4/pull/7559/commits
The first commit is not signed.

And we don't use `git merge` in PR branches. 
Can you rebase?
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#updating-your-branch

If you set up *git* to sign automatically, `git rebase` will sign your all commits.

**Comment by kenjis** [comment]
See https://github.com/codeigniter4/CodeIgniter4/pull/7559/commits
You are merging again and again.

As I said, we do not use `git merge`, but use `git rebase`.

**Comment by BennyBPB** [comment]
> See https://github.com/codeigniter4/CodeIgniter4/pull/7559/commits You are merging again and again.
> 
> As I said, we do not use `git merge`, but use `git rebase`.

Thanks! Sorry, signing and rebase was completely new to me... I think I got it right now.

File: public/index.php [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
 
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
```
Comment (kenjis) [inline public/index.php:11]: Why do you add these comment?

**Comment by kenjis** [comment]
On macOS.

`chdir(FCPATH);`
1.7629029750824 seconds

`if (getcwd() . DIRECTORY_SEPARATOR !== FCPATH) {}`
1.63840508461 seconds

```php
$s = microtime(true);
for ($i = 0; $i <100000; $i++) {
    // Ensure the current directory is pointing to the front controller's directory
    if (getcwd() . DIRECTORY_SEPARATOR !== FCPATH) {
        chdir(FCPATH);
    }
}
echo microtime(true) - $s;
exit;
```

**Comment by kenjis** [comment]
@BennyBPB Thank you for rebasing!

There are three commits but it seems okay with one commit.
If you can squash the commits, do it.

File: public/index.php (hunk 1 above)
Comment (BennyBPB) [reply public/index.php:11]: It was recommended when running: composer cs public/index.php

Comment (kenjis) [reply public/index.php:11]: It is misuse. We have three config files for php-cs-fixer.
Please use `composer cs-fix` and `composer cs` without argument.

The comment is not needed. Please remove.

**Review by kenjis — CHANGES_REQUESTED** [review]
Remove the license comment.

**Comment by BennyBPB** [comment]
> Remove the license comment.

Done, file comment is removed.

**Comment by BennyBPB** [comment]
> @BennyBPB Thank you for rebasing!
> 
> There are three commits but it seems okay with one commit. If you can squash the commits, do it.

Removing the file comment was commit no 4.
I squished them into 1 commit now.

**Review by datamweb — APPROVED** [review]

**Comment by kenjis** [comment]
@BennyBPB Thank you for squishing!

**Review by kenjis — APPROVED** [review]

**Pull request merged**